### ISSUE TITLE
exclude epollreactor from ReactorBuilder on OS X

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5218]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5218|
|Type|enhancement|
|Created|2011-08-03 12:06:38Z|
|Branch|https://github.com/twisted/twisted/tree/osx-skip-poll-5218|

twisted.internet.reactormixins.ReactorBuilder knows how to generate variations of tests for a number of different reactors.  It includes logic for filtering out reactors which couldn't possibly work in the current runtime environment.  It should treat epollreactor as unusable on OS X.

<details><summary>Searchable metadata</summary>

```
trac-id__5218 5218
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_osx_skip_poll_5218 branches/osx-skip-poll-5218
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1312373198000000 1312373198000000
changetime__1314544167000000 1314544167000000
version__None None
owner__exarkun exarkun
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [32474]) Branching to 'osx-skip-poll-5218'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

I did the same thing for `pollreactor` since OS X does not have `select.poll`.

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/osx-skip-poll-5218)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_3"></a>|@jesstess set owner to @exarkun|
|-|-|

"OS X does not have select.poll"

Just to clarify, you mean this in the doesn't-have-a-functional-select.poll sense, and not the there-is-no-such-thing-as-select.poll sense, right? If we're giving up on pollreactor on OS X altogether, then sure, looks good, please merge, although http://twistedmatrix.com/documents/current/core/howto/choosing-reactor.html should probably be updated to reflect that.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

Good catch.  OS X 10.6 and 10.7 ship a Python build which does not even have `select.poll`, though they ship a libc which does have `poll(2)`.  It's true that a custom build of Python on OS X might somehow enable `select.poll` though.  Despite this, I think we should abandon the pretense of poll support on OS X altogether.  It's enough to support the vendor-provided Python runtime, unless someone strongly feels otherwise (ie, is willing to maintain it - presently this means things like fixing PTY support somehow).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [32479]) Talk about select.poll and drop discussion of OS X entirely

refs [#5218](https://github.com/twisted/twisted/issues/5218)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_8"></a>|@jesstess set owner to @exarkun|
|-|-|

Looks great, please merge!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|

(In [32481]) Merge osx-skip-poll-5218

Author: exarkun
Reviewer: jesstess
Fixes: [#5218](https://github.com/twisted/twisted/issues/5218)

Avoid trying to run the poll and epoll `ReactorBuilder`-style tests on OS X,
where they will never work because those APIs are not available on OS X.
